examples/c/Makefile

```diff
@@ -4,6 +4,7 @@ CLANG ?= clang
 LLVM_STRIP ?= llvm-strip
 BPFTOOL ?= $(abspath ../../tools/bpftool)
```

Comment:
we should change this to point into .output directory (and even have bpftool-specific subdirectory there to avoid the overlap with build artifacts from libbpf). I had to do these changes locally to make everything work reliably:

```diff
$ git diff
diff --git a/examples/c/Makefile b/examples/c/Makefile
index b5791820fdb1..baffe903500a 100644
--- a/examples/c/Makefile
+++ b/examples/c/Makefile
@@ -2,10 +2,10 @@
 OUTPUT := .output
 CLANG ?= clang
 LLVM_STRIP ?= llvm-strip
-BPFTOOL ?= $(abspath ../../tools/bpftool)
 LIBBPF_SRC := $(abspath ../../libbpf/src)
 BPFTOOL_SRC := $(abspath ../../bpftool/src)
 LIBBPF_OBJ := $(abspath $(OUTPUT)/libbpf.a)
+BPFTOOL ?= $(abspath $(OUTPUT)/bpftool/bpftool)
 ARCH := $(shell uname -m | sed 's/x86_64/x86/' | sed 's/aarch64/arm64/' | sed 's/ppc64le/powerpc/' | sed 's/mips.*/mips/')
 VMLINUX := ../../vmlinux/$(ARCH)/vmlinux.h
 # Use our own libbpf API headers and Linux UAPI headers distributed with
@@ -47,7 +47,7 @@ clean:
        $(call msg,CLEAN)
        $(Q)rm -rf $(OUTPUT) $(APPS)

-$(OUTPUT) $(OUTPUT)/libbpf:
+$(OUTPUT) $(OUTPUT)/libbpf $(dir $(BPFTOOL)):
        $(call msg,MKDIR,$@)
        $(Q)mkdir -p $@

@@ -60,7 +60,7 @@ $(LIBBPF_OBJ): $(wildcard $(LIBBPF_SRC)/*.[ch] $(LIBBPF_SRC)/Makefile) | $(OUTPU
                    install

 # Build bpftool
-$(BPFTOOL):
+$(BPFTOOL): | $(dir $(BPFTOOL))
        $(call msg,BPFTOOL,$@)
        $(Q)$(MAKE) OUTPUT=$(dir $(BPFTOOL)) -C $(BPFTOOL_SRC)
```

Can you please incorporate these changes and update PR?